Sources/CircularProgress/CircularProgress.swift

```diff
@@ -415,3 +371,60 @@ public final class CircularProgress: NSView {
 		progressLabel.isHidden = !cancelButton.isHidden
 	}
 }
+
+extension CircularProgress:MouseTrackable{
+	public override func hitTest(_ point: NSPoint) -> NSView? {
+		guard let mousePos = self.window?.mouseLocationOutsideOfEventStream else {return nil}
+		let localMousePos = self.convert(mousePos,from:nil)
+		if let path = path {
+			let contains:Bool = path.contains(localMousePos)
+			return contains ? self : nil
+		}
+		return super.hitTest(point)
+	}
+	var path: CGPath? { return CGPath.init(ellipseIn: bounds.insetBy(dx: 20, dy: 20), transform: nil) }
+	override public func updateTrackingAreas() {
+		guard isCancellable else {
+			return
+		}
+		createTrackingArea([.activeAlways, .mouseMoved, .mouseEnteredAndExited])
+		super.updateTrackingAreas()
+	}
+	override public func mouseEntered(with event: NSEvent) {
+		let viewUnderMouse:NSView? = window?.contentView?.hitTest(event.locationInWindow)
+		if hasMouseEntered == false && viewUnderMouse === self {
+			onOver()
+			hasMouseEntered = true
+		}
+		super.mouseEntered(with: event)
+	}
+	override public func mouseExited(with event: NSEvent) {
+		if hasMouseEntered {
+			hasMouseEntered = false
+			onOut()
+		}
+		super.mouseEntered(with: event)
+	}
+	override public func mouseMoved(with event: NSEvent) {
+		super.mouseMoved(with: event)
+		let viewUnderMouse:NSView? = window?.contentView?.hitTest(event.locationInWindow)
+		if hasMouseEntered == false && viewUnderMouse === self {
+			onOver()
+			hasMouseEntered = true
+		}
+		else if hasMouseEntered && viewUnderMouse !== self{
+			onOut()
+			hasMouseEntered = false
+		}
+	}
+	@objc func onOver(){
```

Comment:
Why are these using `@objc`?

Reply:
So you can override methods in extensions. Putting things in extensions avoids MegaControllers. And makes code easier to comprehend. IMO. https://www.natashatherobot.com/using-swift-extensions/  TLDR: While this might not be “traditional”, the super simple use of extensions in Swift can make code super nice and readable.

Reply:
Leaving this be.